packages/purpose-process/src/model/domain/errors.ts

```diff
@@ -137,12 +153,25 @@ export function organizationIsNotTheProducer(
   organizationId: TenantId
 ): ApiError<ErrorCodes> {
   return new ApiError({
-    detail: `Organization ${organizationId} is not allowed to perform the operation`,
+    detail: `Organization ${organizationId} is not allowed to perform the operation because is not the producer`,
     code: "organizationIsNotTheProducer",
     title: "Organization not allowed",
   });
 }
 
+export function organizationIsNotTheDelegatedProducer(
```

Comment:
Nice! If we think it's worthwhile, we could track to improve also agreement validators with similar errors, at the moment there we always return operationNotAllowed